File: EESoft/DataCenter/DataCenter - Server/Forms/FilesCheckerFrm.cs
using System.Drawing;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using System;
using System.IO;

namespace DataCenter_Server
{
	public partial class FilesCheckerFrm
	{
		public FilesCheckerFrm()
		{
			InitializeComponent();
		}
		public ListViewItem[] LVI = new ListViewItem[1];
		public short ForInt;
		public bool AppCantExit;
        public void CBFilter_SelectedIndexChanged(Object sender, EventArgs e)
        {
            if (CBFilter.SelectedIndex == 0)
            {
                int i = LVFolderFile.CheckedItems.Count;
                for (i = LVFolderFile.CheckedItems.Count - 1; i >= 0; i--)
                    LVFolderFile.CheckedItems[i].Remove();
            }
        }
        public void Btnscan_Click(Object sender, EventArgs e)
        {
            Btnscan.Enabled = false;
            LVI = new ListViewItem[1];
            MainModule.FunLib.GetAllFolderFile(MainModule.ProjectsPath);
            if (MainModule.FunLib.FolderFileList.Length == 0)
                PBData.Maximum = 0;
            else
                PBData.Maximum = MainModule.FunLib.FolderFileList.Length - 1;

            LblTotal.Text = MainModule.FunLib.FolderFileList.Length.ToString();
            PBData.Value = 0;
            for (ForInt = 0; ForInt <= MainModule.FunLib.FolderFileList.Length - 1; ForInt++)
            {
                Application.DoEvents();
                LVI[ForInt] = new ListViewItem();
                PBData.Value = ForInt;
                if (ForInt != 0)
                    Lblpercent.Text = Convert.ToString(Convert.ToUInt64((ForInt / Conversion.Val(Convert.ToString(MainModule.FunLib.FolderFileList.Length - 1))) * 100));
                if (MainModule.FunLib.FolderFileList[ForInt].ID.Length > 0)
                {
                    if (MainModule.FunLib.FolderFileList[ForInt].type == FunctionLib.DataCenterServerLib.FolderAndFile.Folder)
                    {
                        if (Directory.Exists(MainModule.FunLib.FolderFileList[ForInt].FilePath))
                        {
                            MainModule.FunLib.FolderFileList[ForInt].Found = true;
                            LVI[ForInt].ForeColor = Color.Green;
                            LVI[ForInt].Checked = true;
                            Lblfound.Text = Convert.ToString((Conversion.Val(Lblfound.Text) + 1));
                        }
                        else
                        {
                            MainModule.FunLib.FolderFileList[ForInt].Found = false;
                            LVI[ForInt].ForeColor = Color.Red;
                            LblNotFound.Text = Convert.ToString((Conversion.Val(LblNotFound.Text) + 1));
                        }
                        LVI[ForInt].Text = (string)(MainModule.FunLib.FolderFileList[ForInt].type.ToString());
                        LVI[ForInt].SubItems.Add(MainModule.FunLib.FolderFileList[ForInt].FilePath);
                        LVI[ForInt].SubItems.Add(MainModule.FunLib.FolderFileList[ForInt].FileName);
                    }
                    else
                    {
                        if (File.Exists((string)(MainModule.FunLib.FolderFileList[ForInt].FilePath)))
                        {
                            MainModule.FunLib.FolderFileList[ForInt].Found = true;
                            LVI[ForInt].ForeColor = Color.Green;
                            LVI[ForInt].Checked = true;
                            Lblfound.Text = Convert.ToString((Conversion.Val(Lblfound.Text) + 1));
                        }
                        else
                        {
                            MainModule.FunLib.FolderFileList[ForInt].Found = false;
                            LVI[ForInt].ForeColor = Color.Red;
                            LblNotFound.Text = Convert.ToString((Conversion.Val(LblNotFound.Text) + 1));
                        }
                        LVI[ForInt].Text = (string)(MainModule.FunLib.FolderFileList[ForInt].type.ToString());
                        LVI[ForInt].SubItems.Add(MainModule.FunLib.FolderFileList[ForInt].FilePath);
                        LVI[ForInt].SubItems.Add(MainModule.FunLib.FolderFileList[ForInt].FileName);
                    }
                    Array.Resize(ref LVI, LVI.Length + 1);
                }
            }
            Array.Resize(ref LVI, LVI.Length - 2 + 1);
            LVFolderFile.Items.AddRange(LVI);
            Btnscan.Enabled = false;
            BtnStart.Enabled = true;
        }
        public void BtnStart_Click(Object sender, EventArgs e)
        {
            MainModule.FunLib.SetRegValue("Done", FunctionLib.DataCenterServerLib.RegKeyValues.DataChacked);
            MainModule.FrmMainForm = new MainForm();
            MainModule.FrmMainForm.Show();
            Hide();
        }
        public void FilesCheckerFrm_FormClosing(object sender, FormClosingEventArgs e)
        {
            if (AppCantExit == false)
                Application.Exit();
        }
	}
}

File: EESoft/DataCenterVB/Backup/DataCenter - Server/Forms/FilesCheckerFrm.vb
Imports System.IO
Public Class FilesCheckerFrm
    Public LVI(0) As ListViewItem
    Public ForInt As Int16
    Public AppCantExit As Boolean = False
    Private Sub CBFilter_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles CBFilter.SelectedIndexChanged
        If CBFilter.SelectedIndex = 0 Then
            Dim i As Integer = LVFolderFile.CheckedItems.Count
            For i = LVFolderFile.CheckedItems.Count - 1 To 0 Step -1
                LVFolderFile.CheckedItems.Item(i).Remove()
            Next
        End If
    End Sub
    Private Sub Btnscan_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Btnscan.Click
        Me.Btnscan.Enabled = False
        ReDim LVI(0)
        FunLib.GetAllFolderFile(ProjectsPath)
        PBData.Maximum = FunLib.FolderFileList.Length - 1
        LblTotal.Text = FunLib.FolderFileList.Length
        PBData.Value = 0
        For ForInt = 0 To FunLib.FolderFileList.Length - 1
            Application.DoEvents()
            LVI(ForInt) = New ListViewItem
            PBData.Value = ForInt
            Lblpercent.Text = CInt((ForInt / Val(FunLib.FolderFileList.Length - 1)) * 100)
            If FunLib.FolderFileList(ForInt).ID.Length > 0 Then
                If FunLib.FolderFileList(ForInt).type = FunctionLib.DataCenterServerLib.FolderAndFile.Folder Then
                    If Directory.Exists(FunLib.FolderFileList(ForInt).FilePath) Then
                        FunLib.FolderFileList(ForInt).Found = True
                        LVI(ForInt).ForeColor = Color.Green
                        LVI(ForInt).Checked = True
                        Lblfound.Text = Val(Lblfound.Text) + 1
                    Else
                        FunLib.FolderFileList(ForInt).Found = False
                        LVI(ForInt).ForeColor = Color.Red
                        LblNotFound.Text = Val(LblNotFound.Text) + 1
                    End If
                    LVI(ForInt).Text = FunLib.FolderFileList(ForInt).type.ToString
                    LVI(ForInt).SubItems.Add(FunLib.FolderFileList(ForInt).FilePath)
                    LVI(ForInt).SubItems.Add(FunLib.FolderFileList(ForInt).FileName)
                Else
                    If File.Exists(FunLib.FolderFileList(ForInt).FilePath) Then
                        FunLib.FolderFileList(ForInt).Found = True
                        LVI(ForInt).ForeColor = Color.Green
                        LVI(ForInt).Checked = True
                        Lblfound.Text = Val(Lblfound.Text) + 1
                    Else
                        FunLib.FolderFileList(ForInt).Found = False
                        LVI(ForInt).ForeColor = Color.Red
                        LblNotFound.Text = Val(LblNotFound.Text) + 1
                    End If
                    LVI(ForInt).Text = FunLib.FolderFileList(ForInt).type.ToString
                    LVI(ForInt).SubItems.Add(FunLib.FolderFileList(ForInt).FilePath)
                    LVI(ForInt).SubItems.Add(FunLib.FolderFileList(ForInt).FileName)
                End If
                ReDim Preserve LVI(LVI.Length)
            End If
        Next
        ReDim Preserve LVI(LVI.Length - 2)
        LVFolderFile.Items.AddRange(LVI)
        Btnscan.Enabled = False
        BtnStart.Enabled = True
    End Sub
    Private Sub BtnStart_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BtnStart.Click
        FunLib.SetRegValue("Done", FunctionLib.DataCenterServerLib.RegKeyValues.DataChacked)
        FrmMainForm = New MainForm
        FrmMainForm.Show()
        Me.Hide()
    End Sub
    Private Sub FilesCheckerFrm_FormClosing(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles Me.FormClosing
        If AppCantExit = False Then
            Application.Exit()
        End If
    End Sub
End Class